test/unit-tests/mobile/android-debug-bridge.ts

```diff
@@ -0,0 +1,181 @@
+import { AndroidDebugBridge } from "../../../mobile/android/android-debug-bridge";
+import { assert } from "chai";
+import { Yok } from "../../../yok";
+import { CommonLoggerStub, ErrorsStub } from "../stubs";
+import { AndroidDebugBridgeResultHandler } from "../../../mobile/android/android-debug-bridge-result-handler";
+const adbPath = "testAdb";
+const adbError = "No space left on device.";
+const adbResponse = "My Cool Response";
+
+describe('androidDebugBridge', () => {
+	let logger: CommonLoggerStub;
+	let adb: any;
+
+	let isSpawnedFromChildProcess = false;
+	let isSpawnedFromEvent = false;
+	let spawnedEvent = "";
+	let spawnedEventOptions = "";
+	let spawnedArgs: string[] = [];
+	let allSpawnedEventsArgs: string[][] = [];
+
+	function setup(options?: {
+		returnError?: boolean
+	}): void {
+		options = options || {};
+		isSpawnedFromEvent = false;
+		isSpawnedFromChildProcess = false;
+		spawnedEvent = "";
+		spawnedEventOptions = "";
+		spawnedArgs = [];
+		allSpawnedEventsArgs = [];
+		createTestInjector(options);
+	}
+
+	function createTestInjector(options?: {
+		returnError?: boolean
+	}): void {
+		options = options || {};
+		const injector = new Yok();
+		injector.register("logger", CommonLoggerStub);
+		injector.register("errors", ErrorsStub);
+		injector.register("config", {});
+		injector.register("options", {});
+		injector.register("staticConfig", {
+			getAdbFilePath: async () => { return adbPath; }
+		});
+		injector.register("androidDebugBridgeResultHandler", AndroidDebugBridgeResultHandler);
+		injector.register("childProcess", {
+			spawnFromEvent: async (command: string, args: string[],
+				event: string, opts?: any): Promise<ISpawnResult> => {
+				isSpawnedFromEvent = command.indexOf(adbPath) !== -1;
+				spawnedArgs = args;
+				allSpawnedEventsArgs.push(args);
+				spawnedEvent = event;
+				spawnedEventOptions = opts;
+
+				return {
+					stderr: options.returnError ? adbError : "",
+					stdout: options.returnError ? "" : adbResponse,
+					exitCode: options.returnError ? 1 : 0
+				};
+			},
+			spawn: async (command: string, args?: string[], opts?: any): Promise<any> => {
+				isSpawnedFromChildProcess = command.indexOf(adbPath) !== -1;
+				spawnedArgs = args;
+
+				return {
+					stderr: options.returnError ? adbError : "",
+					stdout: options.returnError ? "" : adbResponse,
+				};
+			}
+		});
+		adb = injector.resolve(AndroidDebugBridge);
+		logger = injector.resolve("logger");
+	}
+
+	_.each(['executeCommand', 'executeShellCommand'], methodName => {
+		describe(methodName, () => {
+
+			it('should spawn from event by default', async () => {
+				const expectedArgs: string[] = getArgs();
```

Comment:
No need to specify the type of variable. It will be automatically inferred by ts compiler.